llama_index/embeddings/clip.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from typing import Any, List
+
+from PIL import Image
+
+from llama_index.bridge.pydantic import Field, PrivateAttr
+from llama_index.embeddings.base import (
+    DEFAULT_EMBED_BATCH_SIZE,
+    BaseEmbedding,
+    Embedding,
+)
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_CLIP_MODEL = "ViT-B/32"
+
+
+class ClipEmbedding(BaseEmbedding):
```

Comment:
I wonder how this might fit better into the current indexing pipelines.

Likely a future PR, but it seems like there should be a way to track the modalities that an embedding model supports, so that indexing can call the correct model/method to embed a particular data type (text vs. images)

Reply:
@logan-markewich I am making a new complicated pr for multi-modal retrieval (MM vector store). Clip embedding will be the default choice for image encoder.

Reply:
Nice, that makes sense!